provisioner/template.go

```diff
@@ -0,0 +1,102 @@
+package provisioner
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+type applyContext struct {
+	manifestData          map[string]string
+	baseDir               string
+	computingManifestHash bool
+	readTemplate          func(string) ([]byte, error)
+}
+
+func newApplyContext(baseDir string) *applyContext {
+	return &applyContext{
+		baseDir:      baseDir,
+		manifestData: make(map[string]string),
+	}
+}
+
+// applyTemplate takes a fileName of a template and the model to apply to it.
+// returns the transformed template or an error if not successful
+func applyTemplate(context *applyContext, filePath string, data interface{}) (string, error) {
```

Comment:
nit: I would rather call this something like `renderTemplate`. Apply, in my mind, is more in the context of `kubectl apply`.